spec/lib/object_serializer_inheritance_spec.rb

```diff
@@ -93,7 +107,7 @@ class EmployeeSerializer < UserSerializer
     attributes :location
     attributes :compensation
 
-    has_one :account
+    has_one :account, serializer: EmployeeAccountSerializer
```

Comment:
This here was extremely unclear what the behavior should have been.  There is a property called `account`, but the class is defined as `EmployeeAccount`.  Should it assume `AccountSerializer` because of the property name?  Should it read the object and see that it's an `EmployeeAccount` and therefore use `EmployeeAccountSerializer`?

FYI, the existing behavior (and the updated behavior) is that `has_one :account` will resolve to `AccountSerializer`.  This has not changed.  What has changed is that it now requires the serializer to exist.  But, the reality is that you probably won't have an `AccountSerializer` when the class is `EmployeeAccount`.  To me, this looks like a bug here in the `EmployeeSerializer`, where we should expect it to be `EmployeeAccountSerializer` and therefore the relationship would somehow need to specify the serializer.

Reply:
It looks a bit messy, but it seems like the `account` serializer was never defined and actually not used except in two tests (lines 144-151). I think we can just replace the account with something that already has a defined serializer, like move the `has_one :photo` into the `EmployeeSerializer` and update the tests to reflect `photo` instead of the `account`. Plus we delete a couple of extra lines.